### PR TITLE
Support Jib on Google Cloud Build

### PR DESCRIPTION
**Pull request by dgageot**
Requires a new schema version (`v1beta3`)

### CODE REVIEW

**Comment by dgageot** [comment]
Looks like the kokoro failure is not related. Or is it?

File: pkg/skaffold/constants/constants.go [hunk 1]
```diff
@@ -59,6 +59,8 @@ const (
 	UpdateCheckEnvironmentVariable = "SKAFFOLD_UPDATE_CHECK"
 
 	DefaultCloudBuildDockerImage = "gcr.io/cloud-builders/docker"
+	DefaultCloudBuildMavenImage  = "gcr.io/cloud-builders/mvn"
```
Comment (r2d4) [inline pkg/skaffold/constants/constants.go]: WDYT about using the gcr.io/k8s-skaffold/skaffold image for every builder?

Then we wouldn't have to use multiple images

Comment (r2d4) [reply pkg/skaffold/constants/constants.go]: Copy the context + skaffold.yaml, then invoke `skaffold build` and you get all the builders for free.

Comment (dgageot) [reply pkg/skaffold/constants/constants.go]: I don't know about that. I have a few questions:
 + Is skaffold pre-pulled on GCB?
 + How would you build only one artifact with `skaffold build`?
 + Isn't that a bit too inception-ish?

File: pkg/skaffold/build/gcb/jib.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+package gcb
+
+import (
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/jib"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest"
+	cloudbuild "google.golang.org/api/cloudbuild/v1"
+)
+
+// TODO(dgageot): check that `package` is bound to `jib:build`
```
Comment (nkubala) [inline pkg/skaffold/build/gcb/jib.go]: is this still TODO or can it be removed?

Comment (dgageot) [reply pkg/skaffold/build/gcb/jib.go]: It is still a TODO that might or might not be important

**Pull request merged**